gcp_variant_transforms/libs/annotation/vep/vep_runner_test.py

```diff
@@ -93,13 +82,59 @@ def test_get_output_pattern(self):
     output_pattern = self._create_test_instance().get_output_pattern()
     self.assertEqual(output_pattern, _OUTPUT_DIR + '/**_vep_output.vcf')
 
+  def _validate_run_for_all_files(self):
+    matcher = _PartialCommandMatcher(
+        [f[0] for f in test_util.INPUT_FILES_WITH_SIZE])
+    for args_list in self._mock_pipelines.run.call_args_list:
+      self.assertEqual(args_list, mock.call(body=matcher))
+    self.assertEqual(len(matcher.input_file_set), 0)
+
   def test_run_on_all_files(self):
-    test_instance = self._create_test_instance()
+    num_workers = len(test_util.INPUT_FILES_WITH_SIZE) / 2 + 1
+    test_instance = self._create_test_instance(
+        self._get_pipeline_args(num_workers))
     with patch('apache_beam.io.filesystems.FileSystems', _MockFileSystems):
       test_instance.run_on_all_files()
     all_call_args = self._mock_pipelines.run.call_args_list
-    self.assertEqual(len(all_call_args), len(_INPUT_FILES_WITH_SIZE))
-    # TODO(bashir2): Add assertions about call arguments.
+    self.assertEqual(len(all_call_args), num_workers)
+    self._validate_run_for_all_files()
 
-
-# TODO: Add more tests and validations after the first round of review.
+  def test_run_on_all_files_with_more_workers(self):
+    num_workers = len(test_util.INPUT_FILES_WITH_SIZE) + 5
+    test_instance = self._create_test_instance(
+        self._get_pipeline_args(num_workers))
+    with patch('apache_beam.io.filesystems.FileSystems', _MockFileSystems):
+      test_instance.run_on_all_files()
+    all_call_args = self._mock_pipelines.run.call_args_list
+    self.assertEqual(len(all_call_args), len(test_util.INPUT_FILES_WITH_SIZE))
+    self._validate_run_for_all_files()
+
+
+class _PartialCommandMatcher(object):
+  """This is used for checking that calls to Pipelines API cover all inputs.
+
+  We need this matcher to avoid duplicating the whole JSON object created in
+  the production code. Instead we just do a simple heuristic match by going
+  through all `commands` and check if at least in one of them one of the
+  input files appear.
+  """
+
+  def __init__(self, input_file_list):
+    self.input_file_set = set(input_file_list)
+
+  def __eq__(self, other):
```

Comment:
Please note that this pattern can be replaced with a Spy pattern similar to what I have done in [_CounterSpy](https://github.com/googlegenomics/gcp-variant-transforms/blob/b5ae3e74699e5231699d724483f0d0cc2e49f248/gcp_variant_transforms/libs/processed_variant_test.py#L30).

Here I was mostly trying to see how easy/readable it is to achieve my testing goals with mock only. So if it is hard to understand, there are less pythonic mock alternatives too.

Reply:
This is ok. But, could you please make the `__eq__` method idempotent? It's strange how it's changing the state of the class. I think it's more readable to to create a set of `command_part`  elements and then just check whether the two sets intersect.

Reply:
Hmm, not being idempotent is actually the point. So if you do equality comparison on two sets of actions that have the same file (from the input set) the first one should succeed but the second should fail because we don't want to run VEP twice on the same file; note the "or it was repeated" in the error log message below.

Your comment actually makes my concerns stronger about this code not being easy to understand. As I said, I can change it to the Spy pattern which is probably easier to read/understand (For now, I extended class comment a little to make this more clear but good code should be self documenting as much as possible).

Reply:
Done.